### PR TITLE
Named anonymous enums in psp2.

### PR DESCRIPTION
**Pull request by Rinnegatamante**
Named all anonymous enums in psp2 folder (psp2kern is still uncovered, wanted to know if this part is good enough before proceding with psp2kern.)

I've also replaced some multiple defines with new enums (for example certain headers had its specific errorcode list as a list of defines, i replaced them with an enum).

### CODE REVIEW

File: include/psp2/appmgr.h [hunk 1]
```diff
@@ -31,39 +31,43 @@ enum {
...
-enum {
+typedef enum SceAppMgrInfoBarTransparency {
 	SCE_APPMGR_INFOBAR_TRANSPARENCY_OPAQUE	= 0,
 	SCE_APPMGR_INFOBAR_TRANSPARENCY_TRANSLUCENT	= 1
-};
+} SceAppMgrInfoBarTransparency;
 
 typedef struct SceAppMgrSystemEvent {
-	int systemEvent;
+	SceAppMgrSystemEventType systemEvent;
```
Comment (devnoname120) [inline include/psp2/appmgr.h]: Is it safe to put an enum in a struct?

File: include/psp2/appmgr.h [hunk 2]
```diff
@@ -31,39 +31,43 @@ enum {
...
 	uint8_t reserved[60];
 } SceAppMgrSystemEvent;
 
 typedef struct SceAppMgrAppState SceAppMgrAppState; // Missing struct
 typedef struct SceAppMgrExecOptParam SceAppMgrExecOptParam; // Missing struct
 typedef struct SceAppMgrLaunchAppOptParam SceAppMgrLaunchAppOptParam; // Missing struct
 
+typedef struct sceAppMgrLoadExecOptParam {
+	int reserved[256 / 4];
```
Comment (devnoname120) [inline include/psp2/appmgr.h]: Either put 64 or change the type to char and leave 256.

File: include/psp2/message_dialog.h [hunk 3]
```diff
@@ -14,27 +14,22 @@
...
 	//! Illegal parameter
 	SCE_MSG_DIALOG_ERROR_PARAM	= 0x80100a01
-};
+} SceMsgDialogErrorcode;
 
-enum {
-	//! Max length of a user message
-	SCE_MSG_DIALOG_USER_MSG_SIZE	= 512
-};
+#define SCE_MSG_DIALOG_USER_MSG_SIZE	512
```
Comment (devnoname120) [inline include/psp2/message_dialog.h]: The comment got removed, is it intended?

File: include/psp2/npdrmpackage.h [hunk 4]
```diff
@@ -14,7 +14,7 @@ extern "C" {
 #endif
 
 /** Options for _sceNpDrmPackageDecrypt */
-typedef struct {
+typedef struct _sceNpDrmPackageDecrypt{
```
Comment (devnoname120) [inline include/psp2/npdrmpackage.h]: Missing space before `{`.

File: include/psp2/pgf.h [hunk 5]
```diff
@@ -21,14 +21,14 @@ typedef void *SceFontHandle;
 
 /* struct */
 
-enum {
+typedef enum SceFontErrorcode{
```
Comment (devnoname120) [inline include/psp2/pgf.h]: Missing space before `{`.

File: include/psp2/touch.h [hunk 6]
```diff
@@ -13,48 +13,45 @@ extern "C" {
...
-};
+} SceTouchSamplingState;
 
 /**
  * Info field of SceTouchReport structure
  *
  * @see SceTouchReport()
  */
-enum {
+typedef enum SceTouchReportInfo{
```
Comment (devnoname120) [inline include/psp2/touch.h]: Missing space before `{`.

File: include/psp2/touch.h [hunk 7]
```diff
@@ -13,48 +13,45 @@ extern "C" {
...
 	SCE_TOUCH_REPORT_INFO_HIDE_UPPER_LAYER = 0x0001
-};
+} SceTouchReportInfo;
 
 /** Touch error codes */
-enum {
+typedef enum SceTouchErrorcode{
```
Comment (devnoname120) [inline include/psp2/touch.h]: Missing space before `{`.

File: include/psp2/display.h [hunk 8]
```diff
@@ -24,16 +24,18 @@ enum {
...
+typedef enum SceDisplaySetBufSync{
 	/** Buffer change effective immediately */
 	SCE_DISPLAY_SETBUF_IMMEDIATE = 0,
 	/** Buffer change effective next frame */
 	SCE_DISPLAY_SETBUF_NEXTFRAME = 1
-};
+} SceDisplaySetBufSync;
 
 /**
  * Structure used with sceDisplaySetFrameBuf to set/update framebuffer.
```
Comment (devnoname120) [inline include/psp2/display.h]: `::sceDisplaySetFrameBuf `

**Comment by Rinnegatamante** [comment]
Fixed a bunch of reported issues and some non reported too. What about the usage of enums in prototypes/structs?

**Comment by d3m3vilurr** [comment]
Some error code enums ended `Errorcode`(eg. `SceAppUtilErrorcode`). Please change to `ErrorCode`

**Comment by Rinnegatamante** [comment]
@d3m3vilurr Done

File: include/psp2/libssl.h [hunk 9]
```diff
@@ -9,13 +9,15 @@
...
-#define SCE_SSL_ERROR_INVALID_VALUE  0x804351FE
-#define SCE_SSL_ERROR_INVALID_FORMAT 0x80435108
+typedef enum SceSslErrorCode {
+	SCE_SSL_ERROR_BEFORE_INIT    = 0x80435001,
+	SCE_SSL_ERROR_ALREADY_INITED = 0x80435020,
+	SCE_SSL_ERROR_OUT_OF_MEMORY  = 0x80435022,
+	SCE_SSL_ERROR_INTERNAL       = 0x80435026,
+	SCE_SSL_ERROR_NOT_FOUND      = 0x80435025,
+	SCE_SSL_ERROR_INVALID_VALUE  = 0x804351FE,
+	SCE_SSL_ERROR_INVALID_FORMAT = 0x80435108
```
Comment (d3m3vilurr) [inline include/psp2/libssl.h]: can you sort this?

File: include/psp2/usbd.h [hunk 10]
```diff
@@ -13,19 +13,21 @@
...
+	SCE_USBD_ERROR_INTERRUPTED = 0x8024000a,
+	SCE_USBD_ERROR_INVALID_ARG = 0x80240002,
+	SCE_USBD_ERROR_IO = 0x80240001,
+	SCE_USBD_ERROR_NO_DEVICE = 0x80240004,
+	SCE_USBD_ERROR_NO_MEM = 0x8024000b,
+	SCE_USBD_ERROR_NOT_FOUND = 0x80240005,
+	SCE_USBD_ERROR_NOT_SUPPORTED = 0x8024000c,
+	SCE_USBD_ERROR_OVERFLOW = 0x80240008,
+	SCE_USBD_ERROR_PIPE = 0x80240009,
+	SCE_USBD_ERROR_TIMEOUT = 0x80240007
```
Comment (d3m3vilurr) [inline include/psp2/usbd.h]: also this.

**Comment by devnoname120** [comment]
Summary of the discussions on #vitasdk:
- Using `enum` in `struct` has the risk that the resulting `struct` will have an incorrect size (for example with the compilation flag `-fshort-enums`).
- `#pragma enum(4)` could fix the issue, but it might not be supported by all compilers.
- `struct myStruct {MyEnum enum1 : 32;};` would work, but it needs to be put every time an `enum` is used in a `struct`.

**Comment by Rinnegatamante** [comment]
My two cents are for the enums with specified bit sizes.

**Comment by devnoname120** [comment]
I think that `#pragma enum(4) enum {...} #pragma enum(pop)` is the best solution but unit tests need to be added to make sure that it always behaves correctly (on vita's gcc and g++).

**Comment by Rinnegatamante** [comment]
I'm going to revert the enums in structs and add a proper comment for such cases like
`int sample; //<! One of ::MyEnum`

So that the pull can be merged and in the meantime, more tests can be performed in order to get the best solution.

**Comment by d3m3vilurr** [comment]
gcc looks not support`#progma enum`.
so, we should choice another options

1. add bit size to struct `:4`
2. add internal max value to enum `VALUE_MAX = 0xFFFFFFFF`

**Comment by Rinnegatamante** [comment]
I've finished to revert the enums in structs, however i don't know what to do with SceBool (https://github.com/vitasdk/vita-headers/pull/179/files#diff-1764d52c81ce619a0662c175775d7d34L41)

It was defined as 32 bits before so what to do? Add a VALUE_MAX = 0xffffffff or just set SCE_FALSE to 0xFFFFFFFF?

**Comment by devnoname120** [comment]
I agree with doing `int sample; //<! One of ::MyEnum`.

Is this WIP? [Some changes](https://github.com/vitasdk/vita-headers/pull/179/files#diff-897fa7a1cc943f4dc152757e2e276537R45) seem to be missing.

**Comment by Rinnegatamante** [comment]
That struct was like this even before my PR

File: include/psp2/types.h [hunk 11]
```diff
@@ -38,10 +38,11 @@ typedef uint64_t SceULong64;
 typedef unsigned int SceSize;
 typedef int SceSSize;
 
-typedef enum SceBool {
+typedef int SceBool;
+enum SceBool {
 	SCE_FALSE,
 	SCE_TRUE
-} SceBool;
+};
```
Comment (devnoname120) [inline include/psp2/types.h]: `SceBool` is defined twice with different types here.

**Comment by Rinnegatamante** [comment]
Should be good enough now.

**Comment by Rinnegatamante** [comment]
Is there something missing still?

**Comment by Rinnegatamante** [comment]
Bump

File: include/psp2/appmgr.h [hunk 12]
```diff
@@ -13,57 +13,56 @@
...
-	SCE_APPMGR_INFOBAR_TRANSPARENCY_TRANSLUCENT	= 1
-};
+	SCE_APPMGR_ERROR_TOO_LONG_ARGV      = 0x8080201d
+} SceAppMgrErrorCode;
+
+typedef enum SceAppMgrSystemEventType {
+	SCE_APPMGR_SYSTEMEVENT_ON_RESUME             = 0x10000003,
+	SCE_APPMGR_SYSTEMEVENT_ON_STORE_PURCHASE     = 0x10000004,
+	SCE_APPMGR_SYSTEMEVENT_ON_NP_MESSAGE_ARRIVED = 0x10000005,
+	SCE_APPMGR_SYSTEMEVENT_ON_STORE_REDEMPTION   = 0x10000006
```
Comment (d3m3vilurr) [inline include/psp2/appmgr.h]: just a question, why remove tail comma?

Comment (Rinnegatamante) [inline include/psp2/appmgr.h]: Coherency with the other files styling in the SDK.

Comment (d3m3vilurr) [inline include/psp2/appmgr.h]: ok. make sense.

if no one has more review opinions, I'll merge this. just waiting a few more hours ;)

**Pull request merged**